fog/view/server/src/db_fetcher.rs

```diff
@@ -268,110 +229,84 @@ impl<DB: RecoveryDb + Clone + Send + Sync + 'static> DbFetcherThread<DB> {
         let mut has_more_work = false;
 
         // See whats the next block number we need to load for each invocation we are aware of.
-        let (ingestable_ranges, missing_block_ranges) = {
-            let shared_state = self.shared_state();
-            (
-                shared_state.ingestable_ranges.clone(),
-                shared_state.missing_block_ranges.clone(),
-            )
-        };
+        let ingress_keys = self.shared_state().ingress_keys.clone();
 
-        let next_block_index_per_invocation_id = self.block_tracker.next_blocks(&ingestable_ranges);
+        let next_block_index_per_ingress_key = self.block_tracker.next_blocks(&ingress_keys);
```

Comment:
There's something suboptimal here: `next_blocks()` is aware of the last scanned block, but it doesn't do anything with that information. This means the code in this function will repeatedly try to load block data that we actually know is not yet present. I don't think it's a big deal right now but this should probably be documented as potential optimization.

Reply:
:+1: this is fog-426